experiment/benchmark.py

```diff
@@ -136,6 +157,86 @@ def from_yaml(cls, benchmark_path: str) -> List:
                 cppify_headers=cppify_headers,
                 commit=commit,
                 use_context=use_context,
+                is_new_integration=data['is_new_integration'],
+                jvm_special_properties=function.get('jvm_special_properties'),
+                build_project_name=data.get('build_project_name'),
+                function_dict=function))
+
+    return benchmarks
+
+  @classmethod
+  def from_java_data_yaml(cls, benchmark_path: str, project: str,
```

Comment:
I don't think we should have this here.

Rather, can you make it such the auto-building will do the following steps:

1) Find one or more builds that work (can we have multiple builds that work and each have some different characteristics, e.g. may include more/less code in the introspector report?)
2) For the build that works, find a harness that works, it could be empty for that matter, but where the fuzz introspector report includes all the information about the project under analysis
3) use the build that works combined with the harness that works and pass this into the regular processing OFG pipeline, and then let the regular OFG pipeline generate the targets via the existing procedures?

Reply:
So, instead of having a "generate benchmarks from scratch" I'd rather have a : "generate a working build with a primitive benchmark, and use the existing OFG pipeline to take care of the rest"

Reply:
That may have problems. Since we are targeting github repository that we assume does not have previous OSS-Fuzz integration. And thus it cannot go for the OFG pipeline because in the original pipeline, it got to query quite some amount to the FI webapp. Which in this new logic is all done in the benchmark which contains all the necessary information, making the pipeline doesn't need to query to FI during prompt generation. But I do agree that this should be done in a separate location. I will consider how should I relocate and revamp this part to make it easier to follow.

Reply:
Ok. I will revamp the code to consider that.